### PR TITLE
Must gather tier1 tests

### PR DESCRIPTION
**Pull request by AviadP**

### CODE REVIEW

File: tests/manage/cluster/test_must_gather.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+        """
+        Check if directory to store must gather logs already exist
+        and use new directory if so.
+
+        Returns:
+            str: Logs directory
+
+        """
+        index = 1
+        directory = f"/tmp/mg_test{index}"
```
Comment (vasukulkarni) [inline tests/manage/cluster/test_must_gather.py]: please import config and use config.RUN['log_dir']

File: tests/manage/cluster/test_must_gather.py [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+            if must_gather_pods:
+                for pod_to_del in must_gather_pods:
+                    self.ocp_obj.wait_for_delete(resource_name=pod_to_del)
```
Comment (vasukulkarni) [inline tests/manage/cluster/test_must_gather.py]: We are not setting up the pods, why delete them?

File: tests/manage/cluster/test_must_gather.py (hunk 1 above)
Comment (AviadP) [reply tests/manage/cluster/test_must_gather.py]: changed to  utilsocsci_log_path() instead - based on config.RUN['log_dir']

File: tests/manage/cluster/test_must_gather.py (hunk 2 above)
Comment (AviadP) [reply tests/manage/cluster/test_must_gather.py]: it seems that TestManage tear down does not delete this pod in time, so this is a double check

Comment (ebenahar) [reply tests/manage/cluster/test_must_gather.py]: @vasukulkarni if I'm not mistaken, a pod is being created for must gather logs collection. It's being destroyed when must gather is done. So here, we wait for that pod to get deleted so leftover checks will be happy

File: tests/manage/cluster/test_must_gather.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+                    logger.info(f"deleted pods: {pod_to_del}")
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: since you're not actively deleting the must gather pod, this log message and the pod variable name can be confusing. I'd rephrase the pod var to 'must_gather_pod' and the log msg to "Pod {must_gather_pod} got deleted successfully"

File: tests/manage/cluster/test_must_gather.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                    time.sleep(3)
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: please avoid sleep. Try to understand why we need to wait here

File: tests/manage/cluster/test_must_gather.py (hunk 2 above)
Comment (vasukulkarni) [reply tests/manage/cluster/test_must_gather.py]: Thanks, I see that now.

File: tests/manage/cluster/test_must_gather.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+        self.ocp_obj = ocp.OCP()
+
+    @pytest.fixture(autouse=True)
+    def teardown(self, request):
+
+        def finalizer():
+            must_gather_pods = self.ocs.get_pods(label_selector='app=must-gather')
+            logger.info(f"must_gather_pods: {must_gather_pods} ")
+
+            while must_gather_pods:
```
Comment (ebenahar) [inline tests/manage/cluster/test_must_gather.py]: this could lead to an infinite loop.
Please use the TimeoutSampler form instead of while loop

**Comment by AviadP** [comment]
verified locally due to jenkins PR tester being unstable
